File: app/src/main/java/me/devsaki/hentoid/database/converters/InstantConverter.java
package me.devsaki.hentoid.database.converters;

import androidx.annotation.Nullable;

import java.time.Instant;

import io.objectbox.converter.PropertyConverter;

public class InstantConverter implements PropertyConverter<Instant, Long> {
    @Override
    @Nullable
    public Instant convertToEntityProperty(Long databaseValue) {
        if (databaseValue == null) return null;
        return Instant.ofEpochMilli(databaseValue);
    }

    @Override
    @Nullable
    public Long convertToDatabaseValue(Instant entityProperty) {
        return entityProperty == null ? null : entityProperty.toEpochMilli();
    }
}

File: app/src/main/java/me/devsaki/hentoid/database/converters/InstantConverter.kt
package me.devsaki.hentoid.database.converters

import io.objectbox.converter.PropertyConverter
import java.time.Instant

class InstantConverter : PropertyConverter<Instant, Long> {
    override fun convertToEntityProperty(databaseValue: Long?): Instant? {
        return if (databaseValue == null) null else Instant.ofEpochMilli(databaseValue)
    }

    override fun convertToDatabaseValue(entityProperty: Instant?): Long? {
        return entityProperty?.toEpochMilli()
    }
}